src/twisted/trial/_asynctest.py

```diff
@@ -403,3 +403,26 @@ def stop():
         finally:
             results = None
             running.pop()
+
+
+    def _waitUntilAllDisconnected(self, reactor, protocols):
```

Comment:
This is just _horrible_!  Also: genius.

However, I think explaining why it's horrible might be a good idea here.  Link to jml's post https://jml.io/pages/how-to-disconnect-in-twisted-really.html about how to do this the "right" way, explain that it's using timed polling, that this is a workaround for legacy test cases, etc.

Also, there's no reason to put this on `TestCase`, especially given that it's not called from here.  `self` isn't even used in the body of the function.

This should maybe go in `proto_helpers`